webapp/src/components/input.jsx

```diff
@@ -43,11 +55,31 @@ export default class Input extends PureComponent {
         }
     };
 
+    isValid = () => {
+        if (!this.props.required) {
+            return true;
+        }
+        const valid = this.props.value && this.props.value.toString().length !== 0;
+        this.setState({invalidRequired: !valid});
```

Comment:
nit: This line cracks me up (naming). No suggestions, though.

Reply:
Totally agree. Not sure either, maybe just `invalid`?

Reply:
done